### ISSUE TITLE
all_points_membership_vectors doesn't take into account clusterer parameter cluster_selection_epsilon

### ISSUE THREAD

**Issue by gansanay** [scikit-learn-contrib/hdbscan#446]
### Steps to reproduce

Here is a code sample using the digits dataset that demonstrates the issue:

```python
from sklearn import datasets
import umap
import hdbscan

# Build an input dataset
digits = datasets.load_digits()
data = digits.data
fit = umap.UMAP(random_state=42)
lowd = fit.fit_transform(data)

clusterer_eps0 = hdbscan.HDBSCAN(min_samples=4, min_cluster_size=6, cluster_selection_epsilon=0, prediction_data=True).fit(lowd)
n_clusters_eps0 = 1+max(clusterer_eps0.labels_)
not_clustered_eps0 = sum(clusterer_eps0.labels_ < 0)
print('eps = 0 / n_clusters: {}'.format(n_clusters_eps0))
print('eps = 0 / not clustered: {}'.format(not_clustered_eps0))
soft_clusters_eps0 = hdbscan.all_points_membership_vectors(clusterer_eps0)
print('eps = 0 / clusters in all_points_membership_vectors: {}'.format(len(soft_clusters_eps0[0])))

clusterer_eps05 = hdbscan.HDBSCAN(min_samples=4, min_cluster_size=6, cluster_selection_epsilon=0.5, prediction_data=True).fit(lowd)
n_clusters_eps05 = 1+max(clusterer_eps05.labels_)
not_clustered_eps05 = sum(clusterer_eps05.labels_ < 0)
print('eps = 0.5 / n_clusters: {}'.format(n_clusters_eps05))
print('eps = 0.5 / not clustered: {}'.format(not_clustered_eps05))
soft_clusters_eps05= hdbscan.all_points_membership_vectors(clusterer_eps05)
print('eps = 0.5 / clusters in all_points_membership_vectors: {}'.format(len(soft_clusters_eps05[0])))
```

### Expected Result

We could expect the number of clusters in `all_points_membership_vectors` to be the same as the number of different non-noise labels as outputted by the clusterer:

```
eps = 0 / n_clusters: 26
eps = 0 / not clustered: 62
eps = 0 / clusters in all_points_membership_vectors: 26
eps = 0.5 / n_clusters: 13
eps = 0.5 / not clustered: 0
eps = 0.5 / clusters in all_points_membership_vectors: 13
```

### Actual Result

Actually, the reduction in the number of clusters obtained through using nonzero value for `cluster_selection_epsilon` doesn't affect the number of clusters in `all_points_membership_vectors`

```
eps = 0 / n_clusters: 26
eps = 0 / not clustered: 62
eps = 0 / clusters in all_points_membership_vectors: 26
eps = 0.5 / n_clusters: 13
eps = 0.5 / not clustered: 0
eps = 0.5 / clusters in all_points_membership_vectors: 26
```

### Impact

In real-world cases where it is required to assign every data point to a cluster, it would feel natural to use the cluster with maximum probability.

**Comment by pancodia** [comment]
The impacted usecase is the same as mine. I am trying to force the noise points to be one of the existing clusters by assigning to the one with highest probability. Is there a way to at least get the probabilities after considering `cluster_selection_epsilon`?
